source/vibe/core/core.d

```diff
@@ -1587,3 +1592,10 @@ private template needsMove(T)
 	// FIXME: reverse the condition and only call .move for non-copyable types!
 	enum needsMove = is(typeof(T.init.move));
 }
+
+version(VibeLibasyncDriver) {
+	shared static ~this() {
+		import libasync.threads : destroyAsyncThreads;
+		destroyAsyncThreads(); // destroy threads		
+	}
+}
```

Comment:
It follows a fix I made in libasync, where it checks that daemon threads are destroyed when exiting to avoid some invalid memory operations. So everything using a newer version of libasync gets an assertion error on exit unless the threads are destroyed properly. I couldn't destroy them in the driver in any DMD version anyways.